src/job_queue/queue.rs

```diff
@@ -524,22 +560,21 @@ mod tests {
                 });
 
                 // process job and print results.
-                handles.push(job_queue.add_job(job1, DoubleJobPriority::Low)?.result_rx());
-                handles.push(
-                    job_queue
-                        .add_job(job2, DoubleJobPriority::Medium)?
-                        .result_rx(),
-                );
-                handles.push(
-                    job_queue
-                        .add_job(job3, DoubleJobPriority::High)?
-                        .result_rx(),
-                );
+                handles.push(job_queue.add_job(job1, DoubleJobPriority::Low)?);
+                handles.push(job_queue.add_job(job2, DoubleJobPriority::Medium)?);
+                handles.push(job_queue.add_job(job3, DoubleJobPriority::High)?);
             }
 
+            // we can't know exact number of jobs in queue because it is already processing.
+            assert!(job_queue.num_jobs() > 0);
+            assert!(job_queue.num_queued_jobs() > 0);
+
             // wait for all jobs to complete.
             let mut results = futures::future::join_all(handles).await;
 
+            assert_eq!(job_queue.num_jobs(), 0);
```

Comment:
If I recall correctly, we use the convention that the expected value goes on the left-hand side.

Reply:
I will try to remember that.  I always seem to mix it up.

Reply:
reversed order in 7fc22f78b47a08ffe0899442cb35affdf2babb67

Reply:
Frankly, I find our convention a little unintuitive when writing the `assert_eq`, but prefer having a convention.